sherpa/data.py

```diff
@@ -487,14 +492,45 @@ def apply(self, array):
             raise DataErr('mismatch', 'mask', 'data array')
         return array[self.mask]
 
-    def notice(self, mins, maxes, axislist, ignore=False):
+    def notice(self, mins, maxes, axislist, ignore=False, integrated=False):
+        """Select a range.
+
+        Parameters
+        ----------
+        mins : sequence of values
+           The minimum value of the valid range (elements may be
+           None).  When not None, it is treated as an inclusive limit,
+           so points >= min are included.
```

Comment:
Should we explain why this is a sequence (presumably for multi-D data?)

Reply:
So, as far as I understand it, this interface is unlikely to be used by anyone, as the classes that extend the `Data` class (I think that's where this is defined) have their own `notice` call, which has a different interface.

Reply:
Ahh, this is from the `Filter` class, which the `Data` class calls with

```python
    def notice(self, mins, maxes, ignore=False, integrated=False):
        self._data_space.filter.notice(mins, maxes, self.get_indep(),
                                       ignore=ignore, integrated=integrated)
```

and, for example, the `Data1D.notice` method is

```python
   def notice(self, xlo=None, xhi=None, ignore=False):
... docstring ...

        Data.notice(self, (xlo,), (xhi,), ignore)
```

I didn't write this code, I am just trying to document it. I would say that this code is not directly used by the user unless they really go digging through the code, so I am reluctant to spend more time documenting this at this time.

Reply:
I think some of the potential confusion is that this code has to deal with `Data1D` (one axis), `Data1DInt` (two axes, if the low and high values), `Data2D` classes (two axes, but independent), and `Data2DInt` classes [*] (which have two pairs of low/hi axes). This is all handled by a single routine but it does **seem to** require some knowledge by the calling classes (e.g. what the low and high edges are and differences between the nD and nDInt cases). This is all based on me trying to reverse-engineer the code so I may be missing something.

One solution is to just remove this documentation.

[*] we support at the low level but don't really have experience with)

Reply:
There have been some minor doc changes to this method in https://github.com/sherpa/sherpa/pull/1216/commits/73ca7cc38e045b2883205f5a11b497a90e8a49bf but I'm not sure they help

Reply:
There's a few more doc changes in https://github.com/sherpa/sherpa/pull/1216/commits/6b6749b62b51eba4471d0234f20f817103649a1a

Reply:
It's a as good as it gets now and that's good enough. Perfect is the enemy of good.